### DISCUSSION TITLE
What's the best way to implement footstep sounds without having to change references to plugin content?

### DISCUSSION THREAD

**Discussion by epcau** [Sixze/ALS-Refactored#395] [Q&A]
I would like to either override the `AlsAnimNotify_FootstepEffects` and replace it on all animations or make my own instance of `FES_Als_Default` and replace it in the FootstepEffects notifies in animations. 

Both these solutions would include an extensive change in animations that I am reluctant to commit because I am building my framework using ALS Refactored as submodule so that I'm sure I can pull updates all the times I want and keep everything nice and easy to update.

Is there an easy way to customize footsteps logic without having to change references in the plugin itself?

Edit: 
The other solution I can think of now would be to commit changes to the default `FES_Als_Default` table adding my custom metasounds and eventually if that gets changed in the future merge it some way, what do you think?

**Comment by ameaninglessname** [comment]
Maybe you can use core redirector to redirect the class to your derived class.

**Reply by epcau** [reply]
After a little of fiddling around I managed to do it perfectly thanks to core redirector and the snippet of code like below.
This is amazing! I didn't even know about this tool and it's incredible to keep your repo and submodules tidy while developing with freedom. Thanks a lot!

`[CoreRedirects]
+PackageRedirects=(OldName="OldPlugin",NewName="/NewPlugin/"`)

**Reply by dehaotu** [reply]
+1 with this approach. Sorry for replying to old post, I just want to add my experience for the people with similar problem. In my case, I have different characters with distinct set of footsteps (human, robots etc). I ended up creating a derived class from the `UAlsAnimNotify_FootstepEffects` and added core redirect to the `DefaultEngine.ini`. 

One caveat is that for my purpose, I want to assign different footstep effect data assets to the default AnimSqueunce (roughly 50+). I ended up hard coding data asset location in my derived class as defaults, instead of reassigning the values manually.
